### DISCUSSION TITLE
Install with XML file

### DISCUSSION THREAD

**Discussion by wayne-sw** [Open-Shell/Open-Shell-Menu#1011] [Q&A] [answered]
Is there a way to command line install Open Shell **and** reference an XML export with specific settings?

**Comment by Ibuprophen** [comment]
I don't believe so. The xml exports are meant to be imported the same way as it was exported. 

This is the only way I had ever worked with those xml files (personally), maybe @ge0rdi might have some input otherwise. 

~Ibuprophen

**Reply by wayne-sw** [reply]
Would LOVE the ability if it could be added in the future.

**Reply by Ibuprophen** [reply]
This type of feature is something nice to have, but it's unfortunately not possible to do in that way. 

Sorry about this... :( 

~Ibuprophen

**Reply by wayne-sw** [reply]
No worries! Never know unless you ask :)

**Reply by Ibuprophen** [reply]
It's not a problem with me! 👍 

A stupid question is one that isn't asked. 

~Ibuprophen

**Reply by rrygl** [reply]
There may be a workaround: export reg HKEY_CURRENT_USER\SOFTWARE\OpenShell to a .reg file, and "reg.exe /s thatfile.reg" without using .xml.

**Reply by Ibuprophen** [reply]
I wouldn't state that it's a workaround as much as an alternative option for the same thing. The export file basically includes the same information as the registry export in a different type of format. 

It can be very harmful to the OS if someone begins doing things in the registry without knowing exactly what they're doing. 

This is why I (and the other developers here) doesn't like to initiate/direct others to perform tasks in the registry, but will only clarify certain registry questions in hopes that they know what they're doing. 

Please correct me @ge0rdi if I'm wrong here. 👍 

~Ibuprophen

**Reply by wayne-sw** [reply]
Appreciate both answers and @rrygl very much appreciate the registry hint, I should have thought to look there myself. @Ibuprophen appreciate the warning, I'm very familiar with registry import/export and have plenty of systems to test with.

**Reply by blackcrack** [reply]
hmm.. the fear of the other is the most evil, because they they hinder others in order to playfully find out what's going on..
so what's happens if it is there a try out.. and goes wrong.. lost anyone a live ?.. no, it gives only the possible for learning more.. had the geeks in the last century had fear in computers and software, we would never have gotten as far as we are now ! ..  the only what could happens.. lost and murder of datas and therewith the send in nirvana and /dev/null or >NUL in WinNT.. and.. stops the lives.. i believe not .. but.. there be many helpful informations who be later very helpful so it is ever helpful for playing around in kindly way to become in learning way many lessons ... so, stop to say other.. it's may dangerous to make something to not know directly and make therewith maybe fear to don't try out different things because the live is trial and error to have a evolutions and learning , the same makes Child's, they try outs.. and the peoples in the early times of computers does the same... man can only learn from it and i support to make different Regfiles for the own Ordering system but it should not be a foreign System where you an other hurt .. but learning, can you at you own System ;) The same with Repository's.. at the own is is okey but at the foreign ..... it is not the owned ! \*eyewinkle*at Ibu
had i also to lean at a PHP-Repo, had made a similar mistake in the past, i had learn out of it..

best

![Screen 00000](https://user-images.githubusercontent.com/7925717/200157997-0bd07916-dcd5-488d-9662-787f6439ef84.jpg)

**Comment by ge0rdi** [comment, marked as answer]
Hello @wayne-sw .

Currently it is **not possible** to provide XML settings file to installer.
Such improvement is tracked in #251.

You can install `Open-Shell` and then import settings using:
`"C:\Program Files\Open-Shell\StartMenu.exe" -xml <path-to-XML-settings>`

Or you can import registry keys directly, but that's not officially supported way (so if anything goes wrong you are on your own).

**Reply by blackcrack** [reply]
i did not tested, but it is possible for add it in the installer, because this is what he want.. (imh?)

> Is there a way to command line install Open Shell **and** reference an XML export with specific settings?
---------------------------------------------------------------------------------------------------------------  
Almanach  Page ~17 (now)
>What are the command line parameters for the installer? (Unattended / Silence Install)
The ClassicShellSetup.exe installer supports many command line parameters:
 <no command line> - runs the installer normally
 extract32 - extracts the 32-bit MSI
 extract64 - extracts the 64-bit MSI
 help - shows the command line help
 <some msiexec options> - the options are passed to msiexec
> * if the options contain %MSI% (all caps) the token is replaced by the name of the extracted MSI file
 >* if %MSI% is not found, the setup runs "msiexec /i <MSI file> <some msiexec options>"
>
>Examples:
 /qn - runs the installer in quiet mode
 /x %MSI% /qb - uninstalls the product in basic UI level
 /f %MSI% - repairs the product
 /l* log.txt - runs the installer and logs the process in the log.txt file
 APPLICATIONFOLDER="C:\Programs\Classic Shell" - sets the installation folder
 /qn ADDLOCAL=ClassicExplorer - installs only Classic Explorer in quiet mode
 /qn ADDLOCAL=ClassicStartMenu - installs only Classic Start Menu in quiet mode
 ADDLOCAL=ClassicStartMenu,ClassicIE - installs the Classic Start Menu and Classic IE. Note: there must be no space before ClassicIE
 /qb START_MENU_FOLDER=0 - runs in basic mode and skips the creation of a start menu folder
 NOSTART=1 - suppresses the automatic launch of the start menu after install
Examples :
"OpenShellSetup_4_4_143-gmmsuryd.exe" /qn ADDLOCAL=StartMenu
For Startmenu only:
OpenShellSetup_4_4_143-gmmsuryd.exe /qn ADDLOCAL=StartMenu
and with ClassicIE : 
OpenShellSetup_4_4_143-gmmsuryd.exe /qn  ADDLOCAL=StartMenu,ClassicExplorer,ClassicIE,ShellUpdate
and check the /? with the installer !
>  
  
@ge0rdi  is there a possible for add xml or does this works ? :  
OpenShellSetup_X_X_XXX.exe /qn  ADDLOCAL=StartMenu,ClassicExplorer,ClassicIE,ShellUpdate  -xml \<path-to-XML-settings> 

(or could it anyone make possible :blush: ? )

  
best
Blacky
